Code maintenance/46647 bump rails to 7.1

dombesz opened the pull request:
https://community.openproject.org/wp/46647

dombesz committed:
[#46647] Bump Rails to 7.1

- Modified config files:
  - config/application.rb
  - config/environment.rb
  - config/environments/development.rb
  - config/environments/production.rb

https://community.openproject.org/work_packages/46647

dombesz committed:
Update the rest of the config files from bin/rails app:update.

dombesz committed:
Use Rails.application instead of OpenProject::Application

dombesz committed:
Upgrade Paper trail

dombesz committed:
Remove serializer deprecation warning.

dombesz committed:
Add storage test configuration

dombesz committed:
Fix rails bug of using url_helpers outside of controller context

For more info see this discussion: https://github.com/rails/rails/pull/46530#issuecomment-1861391487

dombesz committed:
Fix deprecation warning

dombesz committed:
Fix merge(rewhere:true) deprecation warnings

dombesz committed:
Fix deprecation warnings

dombesz committed:
Change ActiveRecord::SchemaMigration.table_name to AR::Base.connection.schema_migration.table_name

See https://github.com/rails/rails/pull/45908/files#diff-353f4f101b1bb04938d761b6e29c88e0f147084e9e276854a77ae5d9a9a3b2a8

dombesz committed:
Define custom migration classes for Project and Bim::IfcModels::IfcModel to avoid non existing enum type declaration errors.

aaron-contreras commented on modules/bim/db/migrate/20210521080035_update_xkt_to_version8.rb:
```diff
@@ -31,11 +31,20 @@ class UpdateXktToVersion8 < ActiveRecord::Migration[6.1]
 
+  # Note: rails 7.1 breaks the class' ancestor chain, and raises an error, when a class
+  # with an enum definition without a database field is being referenced.
+  # Re-defining the Project class without the enum to avoid the issue.
```
```suggestion
  # Re-defining the IfcModel class without the enum to avoid the issue.
```

aaron-contreras commented on modules/bim/db/migrate/20211011204301_add_column_conversion_status_to_ifc_model.rb:
```diff
@@ -27,15 +27,24 @@
 class AddColumnConversionStatusToIfcModel < ActiveRecord::Migration[6.1]
+  # Note: rails 7.1 breaks the class' ancestor chain, and raises an error, when a class
+  # with an enum definition without a database field is being referenced.
+  # Re-defining the Project class without the enum to avoid the issue.
```
Same here

aaron-contreras reviewed (commented):
Looks good! I believe the specs are failing because using `Rails.application.credentials` now expects to find these secrets in a `credentials.yml` file instead of a `secrets.yml`.

dombesz committed:
Rename secrets.yml to credentials.yml

dombesz committed:
Fix typo

dombesz committed:
Remove warning suppress config

dombesz committed:
Replace deprecated fixture_path with fixure_paths

dombesz committed:
Demodulize controller name for default breadcrumb, to avoid translation missing error

dombesz committed:
Use Rails.application.secret_key_base

dombesz committed:
Remove unused before_action

dombesz committed:
Add our custom deprecator to the Rails deprecators in order to inherit the app deprecators config.

dombesz committed:
Fix settings_helper_specs.rb

dombesz committed:
Fix homescreen_controller_spec.rb

dombesz committed:
Fix repositories_controller_spec.rb

dombesz committed:
Remove unused ProjectsController patch from Costs module.

dombesz committed:
Redefine rails application methods to use Rails.application

dombesz committed:
Change expectation for endless range.

Rails now supports endless ranges

https://guides.rubyonrails.org/7_1_release_notes.html#active-model-notable-changes

dombesz committed:
Add missing status_blocked translation.

dombesz committed:
Fix tabular form builder specs, also scope the localized label inference to the form's object name.

dombesz committed:
Fix auth source translation missing.

It seems we had some residues left from a non existing auth_source field on the User model
that has been renamed to ldap_auth_source.

dombesz committed:
Fix custom field cache translation

dombesz committed:
Fix meeting aaj spec

dombesz committed:
Fix project enabled modules contract spec

dombesz committed:
Simplify fix for settings helper spec

dombesz committed:
Add label_identifier translation key

dombesz committed:
Fix wiki controller specs

dombesz committed:
Fix double render errors in permission specs

dombesz committed:
Add unknown_storage translation key

aaron-contreras committed:
Assert redirect against `response.location` for provider_callback

https://github.com/rails/rails/pull/44554

aaron-contreras committed:
Assert redirect against `response.location` for ensure_connection_flow

https://github.com/rails/rails/pull/44554

dombesz committed:
Fix query schema representer spec

dombesz committed:
Fix the super call on the initializer of the RelationOfTypeColumn class

dombesz committed:
Adding a string to an sql literal implicitly is no longer possible, calling to_s

aaron-contreras commented on app/models/query/results/group_by.rb:
```diff
@@ -163,7 +163,7 @@ def aliased_group_by_sort_order(alias_name, sortable, columns_hash, order = nil)
 
     if order
-      column + " #{order} "
+      column.to_s + " #{order} "
```
Nice find. 👍

toy replied:
Will this work:
```suggestion
      "#{column} #{order} "
```
Also wondering if space is needed at the end:
```suggestion
      "#{column} #{order}"
```

dombesz committed:
Use new mailer config preview path syntax

dombesz committed:
Add missing breadcrumb translations

dombesz committed:
Add missing translation keys

dombesz committed:
Remove inexistent action from before filter on StoragesController

dombesz committed:
Fix Meeting specs

dombesz committed:
Stop raising pending connection errors in ferrum

dombesz committed:
Add missing translation keys

dombesz committed:
Wait for the filters dropdown to be loaded before selecting a filter.

dombesz committed:
Fix toaster dismission not being registered

dombesz committed:
Fix update_cost_report_specs by waiting for page reload to finish

dombesz committed:
Fix Board spec timing issue

dombesz committed:
Improve Filter dropdown loading expectation to cover more cases

dombesz committed:
Improve team planner spec wait_for_loaded method

aaron-contreras commented on modules/team_planner/spec/support/pages/team_planner.rb:
```diff
@@ -343,12 +343,7 @@ def y_center(element)
-
-        sleep 5
-      end
+      expect(page).to have_no_css('.op-team-planner--wp-loading-skeleton')
```
Perhaps extend the wait on this? I've seen this retry twice/thrice quite consistently on the CI logs with the previous implementation so it takes at least 5-10 seconds for it to be gone. Maybe:

```suggestion
      expect(page).to have_no_css('.op-team-planner--wp-loading-skeleton', wait: 10)
```

dombesz committed:
Remove loading indicator wait because there is not indicator on the subproject board, reduce wait time for non-presence expectation.

Reducing the element non-presence wait time speeds up the tests and also makes it pass
despite the flickering that happens while reloading the board.
I left a comment in the spec file to follow up on the flickering reload later.

dombesz committed:
Increase wait time for team planner to reload

Co-authored-by: Aaron Contreras <61627014+aaron-contreras@users.noreply.github.com>

dombesz committed:
Skip waiting for filter loading when the filters are embedded in a modal.

dombesz committed:
Increase retry interval for subproject board spec save

dombesz committed:
More fixes on boards

dombesz committed:
Wait longer for project include component

oliverguenther committed:
Don't move the card by index, but by name

oliverguenther committed:
Set show_exceptions to :none to match previous behavior

This otherwise fails the auth_stages spec https://discuss.rubyonrails.org/t/behaviors-in-action-controllers-raising-error-has-been-changed-in-7-1/83933

dombesz committed:
Change auth_stages feature specs to verify the missing http status code instead of a raised exception.

dombesz committed:
Enable actioncontroller caching in test config.

dombesz committed:
Fix update budget specs.

dombesz committed:
Wait before saving budget specs

dombesz committed:
Revert "Change auth_stages feature specs to verify the missing http status code instead of a raised exception."

This reverts commit f8f9fdd68367261dfa66220d427d247a6497af53.

dombesz committed:
Add edit budget page calculated cost expectations.

dombesz committed:
Fix move work package specs by fixing the context menu opening

dombesz committed:
Fix the my spent time widget spec by waiting for the calendar to be displayed.

dombesz committed:
Wait for ajax query in the project status specs

dombesz committed:
Remove old allow_concurrency = false setting in the test config

cbliard committed:
flaky-fix: Wait for ng-select list to refresh after entering text

Without that, we could click on a stale option and stuck the test.

dombesz committed:
Cache classes and remove null cache store config.

dombesz committed:
Lock the rails version to 7.1.3

dombesz committed:
Correct the config.enable_reloading setting in the test environment

dombesz committed:
Cache Principal.default scope to reduce the number of queries on the Capabilities.default scope.

dombesz committed:
Cache principal query directly in the capabilities scope

toy commented on config/environments/test.rb:
```diff
@@ -26,47 +26,56 @@
   # Spring requires to have the classes reloaded. On the CI or when Spring is
   # disabled, it does not need to happen.
-  config.cache_classes = ENV['CI'].present? || ENV['DISABLE_SPRING'].present?
+  config.enable_reloading = !(ENV['CI'].present? || ENV['DISABLE_SPRING'].present?)
```
Will this be simpler to read?
```suggestion
  config.enable_reloading = %w[CI DISABLE_SPRING].none?{ |name| ENV[name].present? }
```

dombesz committed:
Fix and simplify the update_cost_repor_spec

dombesz committed:
Use the filter loaded expectation on hidden elements for mobile view.

dombesz committed:
Re-use work package filter loaded expectation

dombesz committed:
Revert "Fix move work package specs by fixing the context menu opening"

This reverts commit bdd2aaffc582534c417ad408f423037b547d7126.

dombesz committed:
Make the WorkPacakge::Filters#expect_loaded method to work for mobile view as well

dombesz committed:
Address CR comments

dombesz committed:
Change Board page expect_card to wait for reloading the board column

dombesz committed:
Attempt to wait a bit longer for flaky spec

aaron-contreras commented on Gemfile:
```diff
@@ -39,7 +39,7 @@ gem 'activemodel-serializers-xml', '~> 1.0.1'
 gem 'activerecord-session_store', '~> 2.1.0'
 gem 'ox'
-gem 'rails', '~> 7.0', '>= 7.0.3.1'
+gem 'rails', '~> 7.1.2', '>= 7.1.3'
```
Curious :) Is using this versioning syntax any different than:

```suggestion
gem 'rails', '~> 7.1.3'
```

I understand the current syntax as anything above "7.1.2 and below 7.2" AND "at least 7.1.3" Doesn't the above suggestion say the same a bit more concisely?

dombesz replied:
Yes, it is the same. I updated the rule so we can install any minor release inside 7.1.x that is above 7.1.3.

aaron-contreras commented on app/controllers/repositories_controller.rb:
```diff
@@ -46,7 +46,7 @@ class RepositoriesController < ApplicationController
   before_action :find_project_by_project_id
   before_action :authorize
-  before_action :find_repository, except: %i[edit update create destroy destroy_info]
+  before_action :find_repository, except: %i[update create destroy destroy_info]
```
👍

aaron-contreras replied:
⭐️

aaron-contreras commented on app/models/capabilities/scopes/default.rb:
```diff
@@ -155,6 +155,11 @@ def default_sql_by_non_member_with_anonymous
           WHERE enabled_modules.project_id IS NOT NULL OR "actions".module IS NULL
         SQL_PART
       end
+
+      def principal_sql
+        RequestStore[:capabilities_principal_sql] ||=
+          Principal.visible.not_builtin.not_locked.to_sql
+      end
```
Nice :)

aaron-contreras commented on app/models/members/scopes/not_locked.rb:
```diff
@@ -35,7 +35,7 @@ module NotLocked
         includes(:principal)
           .references(:principals)
-          .merge(Principal.not_locked, rewhere: true)
+          .merge(Principal.not_locked)
```
What was `rewhere: true` doing before? Couldn't find any good answers on this around

dombesz replied:
This post explains it https://www.fastruby.io/blog/rails/upgrades/rails-merge-deprecation.html .
Using `.merge(..., rewhere: true)` makes sure that field conditions on the original query are overwritten by the field conditions on the merge query.

For example doing a merge with having `published: true` on the base query and having `published: false` on the merge query: 
 - Without using `rewhere` the query would be  "published = TRUE AND published = FALSE".
 - With `rewhere` the query is "published = FALSE".
 
Now in rails 7.1, the latter behaviour became the default, so using rewhere is deprecated.

aaron-contreras commented on app/models/project.rb:
```diff
@@ -149,7 +149,6 @@ class Project < ApplicationRecord
 
   friendly_id :identifier, use: :finders
 
-  include ::Scopes::Scoped
```
Nice catch on the duplicate.

aaron-contreras commented on app/workers/mails/mailer_job.rb:
```diff
@@ -41,7 +41,7 @@ class Mails::MailerJob < ApplicationJob
   queue_as { ActionMailer::Base.deliver_later_queue_name }
 
   # Retry mailing jobs three times with exponential backoff
-  retry_on StandardError, wait: :exponentially_longer, attempts: 3
+  retry_on StandardError, wait: :polynomially_longer, attempts: 3
```
```suggestion
  # Retry mailing jobs three times with polynomial backoff
  retry_on StandardError, wait: :exponentially_longer, attempts: 3
  retry_on StandardError, wait: :polynomially_longer, attempts: 3
```

aaron-contreras commented on config/environments/test.rb:
```diff
@@ -75,18 +85,30 @@
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
+
+  # Raise error when a before_action's only/except options reference missing actions
+  config.action_controller.raise_on_missing_callback_actions = true
+
```
⭐️

aaron-contreras commented on config/initializers/filter_parameter_logging.rb:
```diff
@@ -28,5 +28,9 @@
 
-# Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+# Configure parameters to be partially matched (e.g. passw matches password) and filtered from the log file.
+# Use this to limit dissemination of sensitive information.
+# See the ActiveSupport::ParameterFilter documentation for supported notations and behaviors.
+Rails.application.config.filter_parameters += %i[
+  passw secret token _key crypt salt certificate otp ssn
+]
```
Nice

aaron-contreras commented on lib/open_project/static_routing.rb:
```diff
@@ -37,9 +37,11 @@ module StaticRouting
-      include Rails.application.routes.url_helpers
 
       included do
+        # See https://github.com/rails/rails/pull/50403
+        include Rails.application.routes.url_helpers
+
         def default_url_options
```
Interesting order of operations! Thanks for linking the pull request

aaron-contreras commented on modules/boards/spec/features/support/board_page.rb:
```diff
@@ -136,9 +136,13 @@ def expect_not_referencable(list_name, work_package)
                                                   '[data-test-selector="op-wp-single-card--content-subject"]',
-                                                  text: card_title)
+                                                  text: card_title,
+                                                  # Wait only 3 seconds for non-presence expectation
```
I think the comment no longer applies.

```suggestion
                                                  # Don't wait on non-presence expectation
```

aaron-contreras commented on modules/costs/app/controllers/costlog_controller.rb:
```diff
@@ -30,7 +30,6 @@ class CostlogController < ApplicationController
   menu_item :work_packages
   before_action :find_project, :authorize, only: %i[edit new create update destroy]
   before_action :find_associated_objects, only: %i[create update]
-  before_action :find_optional_project, only: %i[report]
```
👍

aaron-contreras commented on modules/dashboards/spec/features/project_status_spec.rb:
```diff
@@ -118,7 +118,7 @@ def add_project_status_widget
 
         # The edit field is toggled and the value saved.
-        expect(page).to have_content('AT RISK')
+        expect(page).to have_content('AT RISK', wait: 5)
```
This is not a request for change, but something to think about amongst all of us. It would be nice to agree on a non-arbitrary long wait time for content that is AJAX-dependent and use a constant throughout the app like:

```ruby
AJAX_DEPENDENT_WAIT = 15.seconds # or whatever number we decide to try
expect(page).to have_content('AT RISK', wait: AJAX_DEPENDENT_WAIT)
```

dombesz replied:
Yeah, this is definitely something to think about. Right now we have a lot of arbitrary waiting times, depending on certain test cases.

aaron-contreras commented on modules/meeting/app/controllers/meeting_agenda_items_controller.rb:
```diff
@@ -35,7 +35,7 @@ class MeetingAgendaItemsController < ApplicationController
   before_action :set_agenda_item_type, only: %i[new create]
   before_action :set_meeting_agenda_item,
-                except: %i[index new cancel_new create author_autocomplete_index]
+                except: %i[new cancel_new create] # author_autocomplete_index
```
```suggestion
                except: %i[new cancel_new create]
```

aaron-contreras commented on modules/meeting/spec/models/meeting_acts_as_journalized_spec.rb:
```diff
@@ -78,7 +78,7 @@
       it 'has the updated_at of the meeting as the lower bound for validity_period and no upper bound' do
         expect(meeting.last_journal.validity_period)
-          .to eql(meeting.reload.updated_at...Float::INFINITY)
+          .to eql(meeting.reload.updated_at...)
```
Nice :)

aaron-contreras commented on modules/reporting/spec/features/support/pages/cost_report_page.rb:
```diff
@@ -87,6 +87,11 @@ def show_loading_indicator(present: true)
       end
     end
 
+    def wait_for_page_to_reload
+      show_loading_indicator
+      show_loading_indicator(present: false)
+    end
+
```
That's clever!

aaron-contreras commented on spec/helpers/settings_helper_spec.rb:
```diff
@@ -36,39 +36,44 @@
+      .to receive(:host_name_writable?)
             .and_return true
+
+    allow(I18n).to receive(:t).and_call_original
+    allow(I18n).to receive(:t).with('setting_field_a').and_return('Field A')
+    allow(I18n).to receive(:t).with('setting_field_b').and_return('Field B')
+    allow(I18n).to receive(:t).with('setting_field').and_return('Field')
```
👍

aaron-contreras commented on spec/lib/api/v3/queries/schemas/query_schema_representer_spec.rb:
```diff
@@ -357,7 +357,11 @@
              Queries::WorkPackages::Columns::PropertyColumn.new(:bogus3),
              Queries::WorkPackages::Columns::RelationToTypeColumn.new(type),
-             Queries::WorkPackages::Columns::RelationOfTypeColumn.new(name: :label_relates_to, sym: :relation1)]
+             Queries::WorkPackages::Columns::RelationOfTypeColumn.new(
+               name: :label_relates_to,
+               sym: :relation1,
+               sym_name: :label_relates_to
+             )]
```
This was quite a bit of detective work! Nice

dombesz replied:
Yep, thanks for helping in the process!🙌

aaron-contreras commented on spec/models/work_package/work_package_acts_as_journalized_spec.rb:
```diff
@@ -379,9 +379,8 @@
-
-          expect(work_package.last_journal.validity_period.last)
-            .to eql(Float::INFINITY)
+          expect(work_package.last_journal.validity_period.end)
+            .to be_nil
```
Your grasp on Journals is way further along than mine. All I'm wondering is if this has any behavioral implication

dombesz replied:
We have a good coverage for testing the aggregation periods so I'm confident that it doesn't change the behaviour. Additionally I verified the database entries and they are also correct.

aaron-contreras reviewed (changes requested):
Hell of a job! 🎸

I mostly just left some cleanup to-dos and questions

```
                     `. ___
                    __,' __`.                _..----....____
        __...--.'``;.   ,.   ;``--..__     .'    ,-._    _.-'
  _..-''-------'   `'   `'   `'     O ``-''._   (,;') _,'
,'________________                          \`-._`-','
 `._              ```````````------...___   '-.._'-:
    ```--.._      ,.                     ````--...__\-.
            `.--. `-`                       ____    |  |`
              `. `.                       ,'`````.  ;  ;`
                `._`.        __________   `.      \'__/`
                   `-:._____/______/___/____`.     \  `
                               |       `._    `.    \
                               `._________`-.   `.   `.___
                                             SSt  `------'`
```

dombesz committed:
Lock rails version to 7.1.x that is above 7.1.3

dombesz committed:
Fix typos

dombesz committed:
Use the expect_and_dismiss_toaster method in work package table specs.

dombesz committed:
Remove typo comment

Co-authored-by: Aaron Contreras <61627014+aaron-contreras@users.noreply.github.com>

dombesz commented:
Thanks for the review @aaron-contreras and @toy 🙌. I believe I addressed your comments, please have a look.

toy commented on config/application.rb:
```diff
@@ -114,7 +114,6 @@ class Application < Rails::Application
 
     # Custom directories with classes and modules you want to be autoloadable.
-    config.enable_dependency_loading = true
     config.paths.add Rails.root.join('lib').to_s, eager_load: true
```
Is this still needed with addition of `autoload_lib`?

toy commented on app/controllers/application_controller.rb:
```diff
@@ -574,7 +574,7 @@ def pick_layout(*args)
 
   def default_breadcrumb
-    label = "label_#{self.class.name.gsub('Controller', '').underscore.singularize}"
+    label = "label_#{self.class.name.demodulize.gsub('Controller', '').underscore.singularize}"
```
I think `controller_name` is available here, so this can be replaced with:
```suggestion
    label = "label_#{controller_name.singularize}"
```

toy commented on app/models/capabilities/scopes/default.rb:
```diff
@@ -155,6 +155,11 @@ def default_sql_by_non_member_with_anonymous
       end
+
+      def principal_sql
+        RequestStore[:capabilities_principal_sql] ||=
```
There are many possible ways of using `RequestStore`, but `RequestStore.fetch` seems to be used more often in the codebase compared with `RequestStore[]`

aaron-contreras reviewed (approved):
Excellent! ⭐️🤌

dombesz committed:
Lock rails version to ~>7.1.3

toy commented on Gemfile:
```diff
@@ -163,7 +163,7 @@ gem 'matrix', '~> 0.4.2'
 gem 'meta-tags', '~> 2.20.0'
 
-gem "paper_trail", "~> 12.3"
+gem "paper_trail", "~> 15.1.0"
```
```suggestion
gem 'paper_trail', '~> 15.1.0'
```

toy commented on app/models/token/hashed_token.rb:
```diff
@@ -46,7 +46,7 @@ def self.hash_function(input)
       # Use a fixed salt for hashing token values.
       # We still want to be able to index the hash value for fast lookups,
       # so we need to determine the hash without knowing the associated user (and thus its salt) first.
-      Digest::SHA256.hexdigest(input + Rails.application.secrets.fetch(:secret_key_base))
```
Was fetch ensuring presence of value which will not be done anymore?

aaron-contreras replied:
I guess it does, but on the same note, doesn't Rails complain either way if this is not present when running with `RAILS_ENV=production`? Or was is it the master key that it complains about?

toy replied:
True, I get ``ArgumentError: Missing `secret_key_base` for 'production' environment, set this string with `bin/rails credentials:edit` (ArgumentError)``

dombesz replied:
Yes, calling `Rails.application.secret_key_base` also validates the presence of the `secret_key_base` and raises the error you mentioned.
```
[1] pry(main)> Rails.application.secret_key_base

From: /.rbenv/versions/3.2.3/lib/ruby/gems/3.2.0/gems/railties-7.1.3/lib/rails/application.rb:654 Rails::Application#validate_secret_key_base:

    652: def validate_secret_key_base(secret_key_base)
    653:   binding.pry
 => 654:   if secret_key_base.is_a?(String) && secret_key_base.present?
    655:     secret_key_base
    656:   elsif secret_key_base
    657:     raise ArgumentError, "`secret_key_base` for #{Rails.env} environment must be a type of String`"
    658:   else
    659:     raise ArgumentError, "Missing `secret_key_base` for '#{Rails.env}' environment, set this string with `bin/rails credentials:edit`"
    660:   end
    661: end
```

toy commented on app/models/user.rb:
```diff
@@ -500,7 +500,7 @@ def self.execute_as(user, &)
   #   - LDAP
   def missing_authentication_method?
-    identity_url.nil? && passwords.empty? && auth_source.nil?
+    identity_url.nil? && passwords.empty? && ldap_auth_source_id.nil?
```
Why not `!ldap_auth_source_id?`?

dombesz replied:
I just followed the convention in the method, but I also think it's more descriptive to call `.nil?`

toy replied:
There are multiple variants in this file, using `ldap_auth_source` and `ldap_auth_source_id`, with `blank?`, `present?` or just checking truthiness, so probably better to unify separately from this PR

toy commented on bin/setup:
```diff
@@ -1,29 +1,33 @@
+  system(*args, exception: true)
+end
+
+FileUtils.chdir APP_ROOT do
```
Unless this is generated there is `Dir.chdir` and there is no need to require fileutils then

toy commented on app/models/queries/work_packages/columns/relation_of_type_column.rb:
```diff
@@ -28,9 +28,8 @@
-    super
-
     self.type = type
+    super(name)
```
Is it an unrelated bug fix?

aaron-contreras replied:
It relates to https://github.com/opf/openproject/pull/14440#discussion_r1471816323

toy commented on spec/requests/api/v3/user/create_user_resource_spec.rb:
```diff
@@ -117,7 +117,7 @@ def send_request
 
           expect(last_response.body)
-            .to be_json_eql('authSource'.to_json)
+            .to be_json_eql('ldapAuthSource'.to_json)
```
Can this change be important?

dombesz replied:
Yep it is important, is has slipped through my radar. Thanks! 🙌

dombesz committed:
Use RequestStore fetch, fix typos in Gemfile.

dombesz committed:
Remove unnecessary path loading config. Fix module naming for Constraints::Enterprise.

dombesz committed:
Revert create user api to use authSource for backwards compatibility.

dombesz committed:
Simplify default_breadcrumb method

dombesz committed:
Use DIR instead of FileUtils in bin/setup script

toy reviewed (approved):
Looks good, as far as I can tell

aaron-contreras commented:
Awesome job my friend! @dombesz
